tests/integration/src/test/java/org/apache/pulsar/tests/integration/utils/DockerUtils.java

```diff
@@ -162,16 +164,26 @@ public static String getContainerIP(DockerClient docker, String containerId) {
         throw new IllegalArgumentException("Container " + containerId + " has no networks");
     }
 
-    public static String getContainerHostname(DockerClient docker, String containerId) {
-        return runCommand(docker, containerId, "hostname").trim();
+    public static ContainerExecResult runCommand(DockerClient docker, String containerId, String... cmd)
+            throws Exception {
+        return runCommand(docker, containerId, false, cmd);
     }
 
-    public static String runCommand(DockerClient docker, String containerId, String... cmd) {
+    public static ContainerExecResult runCommand(DockerClient docker,
+                                                 String containerId,
+                                                 boolean ignoreError,
```

Comment:
minor thing, but i think it'd be cleaner to have a custom exception thrown in the case of a non-zero exit, and have the ContainerExecResult accessible from that exception. As it is is good for now though.

Reply:
good idea. will work on that in the next PR.